### ISSUE TITLE
Some questions about the code of restoring the original image x_0 within a single step

### ISSUE THREAD

**Issue by ShiFangming0823** [rmin2000/WaDiff#5]
Thank you for your wonderful work and pioneering approach! ：)  I am very inspired by the idea of restoring from x<sub>t</sub> to x<sub>0</sub> with a single step and distinguishing between different training steps by timestep _t_ in your proposed approach.

When I was studying your code, I had some minor questions about the code of single-step reversion. Is it the code used in implementing equation (4) in the paper is the following code? What I understand is that what `ori_model_output` and `model_output` return in this code are the noise residuals of the two model outputs for the current timestep prediction, respectively. I may be missing something. Please correct me if my understanding is wrong!

```
if ori_model:
    # Remember to add a time threshold and decode the fingerprinted image
    ori_model_output = ori_model(x_t, self._scale_timesteps(t), **model_kwargs)
    fingerprints = generate_random_fingerprints(x_t.shape[0], wm_decoder.fingerprint_size).to(x_t.device)
    fingerprints = fingerprints * (t <= threshold).view(-1, 1)
    x_t_new = (x_t, fingerprints)
    model_output = model(x_t_new, self._scale_timesteps(t), **model_kwargs)
```
The code is in the line 791 of `guided_diffusion / gaussian_diffusion.py`.

Again, thank you for your wonderful work and looking forward to your early reply! Merry Christmas! :)

**Comment by rmin2000** [comment]
Hi, thanks for your interest :)

Yes, you are correct. There is a discrepancy between my public code and the local code that has been altered from L2 norm loss to perceptual loss, and I've updated it accordingly. Again, thanks for pointing out this, and please let me know if there are further issues. Merry Christmas!
